Tests/hypergraph_constructors.py

```
@@ -0,0 +1,369 @@
+            [0, 1]]
+        )
+        HG = Hypergraph(incidence_matrix=incidence_matrix)
+        HG.nodes['key'] = ['a', 'b', 'c', 'd']
+        HG.add_node(properties={'key':'e', 'value':'z'})
+
+        node_df = pd.DataFrame({'Nodes': [0,1,2,3,4],
+                                'key':['a','b','c','d','e'],
+                                'value':[pd.NA, pd.NA, pd.NA, pd.NA, 'z']})
+        assert_frame_equal(node_df, HG.nodes, check_dtype=False)
+
+    def test_add_node_3(self):
```

Ruff (0.8.2)

Redefinition of unused `test_add_node_3` from line 259

(F811)